util/treecache/treecache.go

```diff
@@ -243,7 +264,9 @@ func (tc *ZookeeperTreeCache) recursiveNodeUpdate(path string, node *zookeeperTr
 		}
 	}
 
+	tc.wg.Add(1)
 	go func() {
+		defer tc.wg.Done()
 		numWatchers.Inc()
 		// Pass up zookeeper events, until the node is deleted.
 		select {
```

Comment:
Hmm, are we missing a for loop here?

Reply:
Are you talking about a for loop around the select statement?

I don't think a for loop is needed here because `childWatcher` and `dataWatcher` will only fire once. The `loop` function takes care of re-watching the zk nodes.